kafka_consumer/tests/python_client/test_unit.py

```diff
@@ -43,36 +57,41 @@ def test_tls_config_ok(check, kafka_instance_tls):
 
 
 @pytest.mark.parametrize(
-    'sasl_oauth_token_provider, expected_exception',
+    'sasl_oauth_token_provider, expected_exception, mocked_admin_client',
     [
         pytest.param(
             {},
-            pytest.raises(AssertionError, match="sasl_oauth_token_provider required for OAUTHBEARER sasl"),
+            pytest.raises(Exception, match="sasl_oauth_token_provider required for OAUTHBEARER sasl"),
+            None,
             id="No sasl_oauth_token_provider",
         ),
         pytest.param(
             {'sasl_oauth_token_provider': {}},
-            pytest.raises(ConfigurationError, match="The `url` setting of `auth_token` reader is required"),
+            pytest.raises(Exception, match="The `url` setting of `auth_token` reader is required"),
+            None,
             id="Empty sasl_oauth_token_provider, url missing",
         ),
         pytest.param(
             {'sasl_oauth_token_provider': {'url': 'http://fake.url'}},
-            pytest.raises(ConfigurationError, match="The `client_id` setting of `auth_token` reader is required"),
+            pytest.raises(Exception, match="The `client_id` setting of `auth_token` reader is required"),
+            None,
             id="client_id missing",
         ),
         pytest.param(
             {'sasl_oauth_token_provider': {'url': 'http://fake.url', 'client_id': 'id'}},
-            pytest.raises(ConfigurationError, match="The `client_secret` setting of `auth_token` reader is required"),
+            pytest.raises(Exception, match="The `client_secret` setting of `auth_token` reader is required"),
+            None,
             id="client_secret missing",
         ),
         pytest.param(
             {'sasl_oauth_token_provider': {'url': 'http://fake.url', 'client_id': 'id', 'client_secret': 'secret'}},
-            pytest.raises(Exception, match="NoBrokersAvailable"),  # Mock the expected response after library migration
+            does_not_raise(),
+            mock.MagicMock(),
```

Comment:
To reviewer: I am mocking the `AdminClient` here since otherwise the actual `AdminClient` will attempt to connect to `http://fake.url` to retrieve an OIDC token. The purpose of this test is mostly to make sure the correct Exception+message is raised or not raised.